### ISSUE TITLE
Connection from browser to BP Client fails

### ISSUE THREAD

**Issue by zfi** [parallaxinc/BlocklyProp-cdn#140]
The browser is unable to connect to the BlocklyProp Client after an initial connect is successful. Browser console messages indicate that the web socket is not initialized and the connection fails to open when called.

**Comment by zfi** [comment]
An update from a recent commit included that referenced a projectData['board'] object but the object is undeclared at that point in the code execution. Rolling back a portion of the commit to remove the code added to the blocklypropclient.js and submitting update to the Demo server to further evaluate the update.

**Comment by MatzElectronics** [comment]
Okay, this has to be done carefully, because it interacts with UI elements
that have changed as well.  Best thing to do would be to move this whole
chunk of code to editor.js inside of the $(document).ready() block where
that variable is available.

**Comment by PropGit** [comment]
This is a duplicate of [Blockly Prop Issue #1687](https://github.com/parallaxinc/BlocklyProp/issues/1687).

The WebSocket message is not related, I believe, because BP Client doesn't use Web Sockets but the Web Client is looking for both BlocklyProp Client and BlocklyProp Launcher and is just reporting that web socket connection (to BP C) failed.  The issue linked above indicates the error message that occurs when the connection truly fails to BP C; and it fails because the code @zfi indicated indeed checks for variable contents which sometimes are not available yet (and sometimes are).

**Comment by zfi** [comment]
The patch in PR #141 appears to be effective in the Demo system. We can move the remainder of the patch to the editor.js file to clear up the UI changes based on the board type.

More concerning is the possibility that the projectData global variable does come into scope in an consistent manner or that the code that relies on it is not testing for it before assuming that it is available. We need to address that is this actually is a possible scenario.

**Comment by MatzElectronics** [comment]
Agreed.  Some larger refactoring across blocklypropclient.js, blocklyc.js,
editor.js, and propc.js should help this considerably.

**Comment by MatzElectronics** [comment]
The variable can be tested first, before a comparison.  If it doesn't exist, it leaves the UI alone, which is okay.  I've submitted that change.

**Comment by PropGit** [comment]
This issue appears to be solved.  See Issue #1696.

The current solution may need some work to make it very solid, as mentioned above.
